pkgs/build-support/node/build-npm-package/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchNpmDeps, callPackage, nodejs }:
+
+{ name ? "${args.pname}-${args.version}"
+, src ? null
+, srcs ? null
+, sourceRoot ? null
+, patches ? [ ]
+, npmDepsHash ? ""
+, nativeBuildInputs ? [ ]
+, buildInputs ? [ ]
+, ...
+} @ args:
+
+assert npmDepsHash != "";
+
+let
+  npmDeps = fetchNpmDeps {
+    inherit src srcs sourceRoot patches;
+    name = "${name}-npm-deps";
+    hash = npmDepsHash;
+  };
+
+  # TODO: how do we expose these like Rust? `npmHooks`?
+  inherit (callPackage ./hooks { inherit nodejs; }) npmConfigHook npmBuildHook npmInstallHook;
```

Comment:
I'm not really sure what the best name for exposing the hooks would be, which is why I didn't do it here.